activejob/lib/active_job/serializers/duration_serializer.rb

```diff
@@ -4,12 +4,19 @@ module ActiveJob
   module Serializers
     class DurationSerializer < ObjectSerializer # :nodoc:
       def serialize(duration)
-        super("value" => duration.value, "parts" => Arguments.serialize(duration.parts))
+        super("value" => duration.value, "parts" => Arguments.serialize([duration.parts]))
       end
 
       def deserialize(hash)
         value = hash["value"]
-        parts = Arguments.deserialize(hash["parts"])
+        deserialized_parts = Arguments.deserialize(hash["parts"])
+        parts = if deserialized_parts.first.is_a?(Hash)
+          deserialized_parts.first
+        else
+          # 'parts' used to be serialized like [:years, 1], but it should have been [{ years: 1 }].
+          # This maintains deserialization-compatibility with any old serialized data:
```

Comment:
Let's add a comment to remove this in Rails 7.1.0.

Reply:
Done